test/python/pulse/test_pulse_lib.py

```diff
@@ -106,38 +106,71 @@ def test_sampled_pulse(self):
         sample_pulse = gauss.get_sample_pulse()
         self.assertIsInstance(sample_pulse, Waveform)
         pulse_lib_gaus = gaussian(duration=25, sigma=4,
-                                  amp=0.5j, zero_ends=False).samples
+                                  amp=0.5j, zero_ends=True).samples
         np.testing.assert_almost_equal(sample_pulse.samples, pulse_lib_gaus)
 
     def test_gauss_samples(self):
         """Test that the gaussian samples match the formula."""
         duration = 25
         sigma = 4
         amp = 0.5j
-        # formulaic
-        times = np.array(range(25), dtype=np.complex_)
-        times = times - (duration / 2) + 0.5
+
+        # formulaic (with fixed zeroed endpoints at ``t=-1, duration+1``)
+        times = -1.0 + np.array(range(duration+2), dtype=np.complex_)
+        times -= duration/2
+        times[1:duration+1] += 0.5
+        times[duration+1] += 1  # fix duration+1 endpoint
+
         gauss = amp * np.exp(-(times / sigma)**2 / 2)
+        zero_offset = gauss[0]
+        gauss -= zero_offset
+        amp_scale_factor = amp/(amp-zero_offset) if amp-zero_offset != 0 else 1
+        gauss *= amp_scale_factor
+
+        # verify zero offset and rescaling
+        self.assertEqual(gauss[0], 0.0)
```

Comment:
Why are you testing the samples you created in this test?, you should be testing `samples`. Then almost all of the above is unnecessary. Use `library.continuous.gaussian` to generate this samples should be fine. These comments apply to the tests below as well.

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/4640/commits/8f427d9c326edf952247105d5392d67581d8bc5d.